src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -21,6 +21,52 @@
 #include "ray/gcs/pb_util.h"
 #include "ray/stats/stats.h"
 
+namespace {
+const ray::rpc::ActorDeathCause GenRuntimeEnvFailedCause(const std::string &error_msg) {
+  ray::rpc::ActorDeathCause death_cause;
+  death_cause.mutable_runtime_env_failed_context()->set_error_message(error_msg);
+  return death_cause;
+}
+
+const ray::rpc::ActorDeathCause GenNodeDiedCause() {
+  ray::rpc::ActorDeathCause death_cause;
+  death_cause.mutable_node_died_context()->set_error_message(
+      "The actor is dead because the node where actor resides is dead.");
+  return death_cause;
+}
+
+const ray::rpc::ActorDeathCause GenWorkerDiedCause() {
+  ray::rpc::ActorDeathCause death_cause;
+  death_cause.mutable_worker_died_context()->set_error_message(
+      "The actor is dead because the worker process where actor resides is dead.");
+  return death_cause;
+}
+const ray::rpc::ActorDeathCause GenOwnerDiedCause(const WorkerID &owner_id) {
+  ray::rpc::ActorDeathCause death_cause;
+  ray::rpc::ActorDeathOwnerDiedContext owner_died_context;
+  owner_died_context.set_owner_id(owner_id.Binary());
+  owner_died_context.set_error_message(
+      "The actor is dead because the owner of the actor is dead.");
+  death_cause.mutable_owner_died_context()->Swap(&owner_died_context);
+  return death_cause;
+}
+
+const ray::rpc::ActorDeathCause GenKilledByApplicationCause() {
+  ray::rpc::ActorDeathCause death_cause;
+  death_cause.mutable_killed_by_app_context()->set_error_message(
+      "The actor is dead because it is killed by ray.kill.");
```

Comment:
was killed

Reply:
What about was killed by `ray.kill`? Was killed seems to be a bit ambiguous